Source/Core/DolphinQt/Debugger/CodeViewWidget.cpp

```diff
@@ -587,6 +598,90 @@ void CodeViewWidget::OnContextMenu()
   Update();
 }
 
+void CodeViewWidget::AutoStep(CodeTrace::AutoStop option)
+{
+  // Autosteps and follows value in the target (left-most) register. The Used and Changed options
+  // silently follows target though reshuffles in memory and registers and stops on use or update.
+
+  CodeTrace code_trace;
+  bool repeat = false;
+  std::tuple<u32, bool, bool> trace_status;
+  const auto& [count, empty_trackers, timed_out] = trace_status;
+  std::tuple<std::vector<std::string>, std::set<u32>> tracked;
+  const auto& [reg_tracked, mem_tracked] = tracked;
+
+  QMessageBox msgbox(QMessageBox::NoIcon, tr("Run until"), tr(""), QMessageBox::Cancel);
+  QPushButton* run_button = msgbox.addButton(tr("Keep Running"), QMessageBox::AcceptRole);
+  // Not sure if we want default to be cancel. Spacebar can let you quickly continue autostepping if
+  // Yes.
+
+  do
+  {
+    // Run autostep then update codeview
+    trace_status = code_trace.AutoStepping(repeat, option);
+    emit Host::GetInstance()->UpdateDisasmDialog();
+    repeat = false;
+
+    // Invalid instruction, 0 means no step executed.
+    if (count == 0)
+      return;
+
+    if (empty_trackers)
+    {
+      QMessageBox::warning(
+          this, tr("Overwritten"),
+          tr("Target value was overwritten by current instruction.\nInstructions executed:  ") +
+              QString::number(count),
+          QMessageBox::Cancel);
+      return;
+    }
+
+    tracked = code_trace.GetTracked();
+
+    // Can keep running and try again after a time out.
+    if (timed_out)
+      msgbox.setText(
+          tr("<font color='#ff0000'>AutoStepping timed out. Current instruction is irrelevant."));
+    else
+      msgbox.setText(tr("Value tracked to current instruction."));
+
+    // Mem_tracked needs to track each byte individually, so a tracked word-sized value would have
+    // four entries. The displayed memory list needs to be shortened so it's not a huge list of
+    // bytes. Assumes adjacent bytes represent a word or half-word and removes the redundant bytes.
+    std::set<u32> mem_list;
+    auto iter = mem_tracked.begin();
+
+    while (iter != mem_tracked.end())
+    {
+      const u32 address = *iter;
+      mem_list.insert(address);
+
+      for (u32 i = 1; i <= 3; i++)
+      {
+        if (!mem_tracked.count(address + i))
+          break;
```

Comment:
Sorry, I threw it together and it is a bit confusing. If it has 80000000 and 80000002 then they should both print, because they are separated, so it breaks at 80000001 not being present.